### PR TITLE
modified csminit to put model info in a separate group

### PR DESCRIPTION
**Pull request by jessemapel**
## Description

This moved the user info about the csm model into a separate group called CsmInfo to make clean up easier. It also adds more info to the labels and updates the test CSM models to be geometric models and not just models.

## Related Issue

csm support in ISIS project

## Motivation and Context

This makes it much easier to clean-up what csminit adds to the label in future runs of csminit or spiceinit. A PR with modification to make them clean-up each-other is coming after this.

## How Has This Been Tested?

Tested all csminit tests on Mac and Ubuntu.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/tests/AlternativeTestCsmModel.cpp [hunk 1]
```diff
@@ -4,8 +4,27 @@
 #include <nlohmann/json.hpp>
 using json = nlohmann::json;
 
+const std::string AlternativeTestCsmModel::SENSOR_MODEL_NAME = "TestCsmModelName";
```
Comment (krlberry) [inline isis/tests/AlternativeTestCsmModel.cpp]: Update to AlternativeTestCsmModelName.

File: isis/tests/AlternativeTestCsmModel.cpp [hunk 2]
```diff
@@ -4,8 +4,27 @@
...
+const std::vector<std::string> AlternativeTestCsmModel::PARAM_NAMES = {
```
Comment (krlberry) [inline isis/tests/AlternativeTestCsmModel.cpp:8]: Not sure if it matters here, but in the Plugin createModelFromISD and the test examples, I had this one have TestParams1->3, not just 1 and 2.

File: isis/tests/TestCsmModel.cpp [hunk 3]
```diff
@@ -4,8 +4,35 @@
 #include <nlohmann/json.hpp>
 using json = nlohmann::json;
 
+const std::string TestCsmModel::SENSOR_MODEL_NAME = "TestCsmModelName";
+const std::vector<std::string> TestCsmModel::PARAM_NAMES = {
+  "TestParam1",
```
Comment (krlberry) [inline isis/tests/TestCsmModel.cpp]: Update this or Plugin and example ISD to match? I had TestCsmModel only have TestParam1 and TestParam2

File: isis/tests/FunctionalTestsCsminit.cpp [hunk 4]
```diff
@@ -80,106 +95,154 @@ TEST_F(CSMPluginFixture, CSMInitDefault) {
...
+  ASSERT_EQ(infoGroup["ModelParameterTypes"][0].toStdString(), "FICTITIOUS");
+  ASSERT_EQ(infoGroup["ModelParameterTypes"][1].toStdString(), "REAL");
+  ASSERT_EQ(infoGroup["ModelParameterTypes"][2].toStdString(), "FIXED");
 }
 
 TEST_F(CSMPluginFixture, CSMinitRunTwice) {
-  // Run csminit twice in a row with the same options each time. Verify that end result is 
-  // as if csminit were only run once. 
+  // Run csminit twice in a row. Verify that end result is as if csminit were
+  // only run once with the second arguments.
```
Comment (krlberry) [inline isis/tests/FunctionalTestsCsminit.cpp:134]: :+1: Better wording, thanks!

**Review by krlberry — COMMENTED** [review]
Looks good! Glad you added all the `csm::GeomtericModel` stuff. I made a couple of minor comments, mostly about matching `PARAM_NAMES` with the example ISDs in the test and the ISD parsing in the plugin.

File: isis/tests/AlternativeTestCsmModel.cpp [hunk 5]
```diff
@@ -65,11 +91,21 @@ std::string AlternativeTestCsmModel::getReferenceDateAndTime() const {
...
+  state["test_param_one"] = m_param_values[0];
+  state["test_param_two"] = m_param_values[1];
+  state["test_param_three"] = m_param_values[2];
+  state["test_param_four"] = m_param_values[3];
+  return AlternativeTestCsmModel::SENSOR_MODEL_NAME + "\n" + state.dump();
 }
 
 void AlternativeTestCsmModel::replaceModelState(const std::string& argState) {
-  m_modelState = argState;
+  // Get the JSON substring
```
Comment (jessemapel) [inline isis/tests/AlternativeTestCsmModel.cpp:102]: Technically, we should also check that the string before the new line matches the model name, but I think that's a bit too far.

**Comment by jessemapel** [comment]
@kberryUSGS Okay this should be good now

**Review by krlberry — APPROVED** [review]
Looks great!

**Pull request merged**